eisbuk-admin/cypress/support/commands.js

```diff
@@ -36,28 +37,43 @@ connectFunctionsEmulator(functions, "localhost", 5001);
 
 /**
  * Set up app for testing and log in as default admin
+ * @param {boolean} doLogin - whether to log in as default admin
  */
-Cypress.Commands.add("initAdminApp", () => {
+Cypress.Commands.add("initAdminApp", async (doLogin = true) => {
   // create a random organization name in order to run each test
   // against it's own organization, using the same db without conflicts
   const organization = uuidv4();
   cy.on("window:before:load", (win) => {
     // the `organization` name is set to local storage and read from the app while testing
     win.localStorage.setItem("organization", organization);
   });
-
-  // create or login a default user in order to run the tests with admin access
-  createUserWithEmailAndPassword(
-    auth,
-    defaultUser.email,
-    defaultUser.password
-  ).catch(() => {
-    signInWithEmailAndPassword(auth, defaultUser.email, defaultUser.password);
-  });
-
   // create a default organization (containing the default user as an admin)
-  httpsCallable(
+  await httpsCallable(
     functions,
     CloudFunction.CreateOrganization
-  )({ organization }).catch();
+  )({ organization });
+
+  // Always create a user (and maybe log them out)
+  try {
+    await createUserWithEmailAndPassword(
+      auth,
+      defaultUser.email,
+      defaultUser.password
+    );
+    if (!doLogin) {
+      await signOut(auth);
+    }
+  } catch (error) {
+    if (doLogin) {
+      await signInWithEmailAndPassword(
+        auth,
+        defaultUser.email,
+        defaultUser.password
+      );
+      console.log(`Logged in as ${defaultUser.email}`);
+    } else {
+      await signOut(auth);
+      console.log(`Logged out`);
+    }
```

Comment:
This seems unnecessary as the user will not be signed in if the create user failed. We should only run login if needed (to log in a user who, obviously, exists)

Reply:
That was my thought as well, but I was not able to make it work that way.
I wonder if cypress is not cleaning everything up between tests.
Or maybe I saw something that was not there.
If it works without this, feel free to take it out.

Reply:
Have left this.